service/raft/group0_state_machine.cc

```diff
@@ -139,6 +139,7 @@ static future<> write_mutations_to_database(storage_proxy& proxy, gms::inet_addr
 }
 
 future<> group0_state_machine::merge_and_apply(group0_state_machine_merger& merger) {
+  return async([this, &merger] {
```

Comment:
I don't like this thread proliferation.

Can't we use non-thread gentle functions?

Reply:
I'll see what it takes to generate async serialization code, with which we can have gentle freeze / make_canonical_mutation implementations that don't require a thread.